### PR TITLE
FIREFLY-1527: Update the package/build configuration and cleanup

### LINKED ISSUE #51: pkg_resources is deprecated
The `__init__.py` has the following code:

```python
__version__ = pkg_resources.get_distribution("firefly_client").version
```
This issues a deprecation warning now because `pkg_resources` is deprecated.

The modern equivalent is to use `importlib.metadata.version`:

```python
from importlib.metadata import PackageNotFoundError, version

__version__: str

try:
    __version__ = version("firefly_client")
except PackageNotFoundError:
    # package is not installed
    __version__ = "0.0.0"
```

### LINKED ISSUE #60: undeclared dependencies causing downstream CI errors
I'm running into the undeclared dependency of setuptools when testing notebooks that require this package: https://github.com/Caltech-IPAC/irsa-tutorials/actions/runs/10188864265/job/28185848617#step:5:393

However, addressing https://github.com/Caltech-IPAC/firefly_client/issues/51 would solve this problem, too.

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[1], line 6
      4 from astropy.table import QTable
      5 from astropy.utils.data import download_file
----> 6 from firefly_client import FireflyClient
      7 import numpy as np
      8 import pyvo

File ~/work/irsa-tutorials/irsa-tutorials/.tox/py312-test/lib/python3.12/site-packages/firefly_client/__init__.py:5
      3 from .env import Env
      4 from .range_values import RangeValues
----> 5 import pkg_resources
      7 __version__ = pkg_resources.get_distribution("firefly_client").version

ModuleNotFoundError: No module named 'pkg_resources'

_______________ tutorials/firefly/SEDs_in_Firefly.ipynb::Cell 4 ________________
```

### PR DESCRIPTION
**Pull request by jaladh-singhal**
Fixes [FIREFLY-1527](https://jira.ipac.caltech.edu/browse/FIREFLY-1527), fixes #60, fixes #51 

- Fixed version retrieval in code, as suggested by #51
- Replaced `setup.py` and `setup.cfg` with `pyproject.toml`, why?
  - [PyPA](https://www.pypa.io/en/latest/) recommends it, see the motivation behind this change [summarised at setuptools docs](https://setuptools.pypa.io/en/latest/build_meta.html#what-is-it)
  - [jupyter_firefly_extensions](https://github.com/Caltech-IPAC/jupyter_firefly_extensions/blob/master/pyproject.toml) is already using this modern config, this package should also
- Cleaned up obsolete files like travis.yml (we now have a github action to build the website) and requirements.txt (because we define dependencies in pyproject.toml)
- Updated docs pipeline to use optional `[docs]` dependencies 
- Fixed some logical linting errors in source files
- Updated release documentation as per build config changes

## Testing
Checkout this branch locally and then in a fresh local venv or conda env, doing `pip install .` should install firefly_client. Then run example notebooks - they should work

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,3 @@
-requests~=2.30.0
-setuptools~=41.0.1
-websocket-client~=0.58.0
-
 # Documentation dependencies
 Sphinx~=7.1.0
```
Comment (bsipocz) [inline requirements.txt]: you should also add these as `[docs]` in `optional-dependencies`

Comment (jaladh-singhal) [reply requirements.txt]: Good idea! Done - the toml syntax is really cool

**Comment by bsipocz** [comment]
The notebooks now pass all the tests with OSX, as well as without installing setuptools, so from my end this looks good.

**Review by robyww — APPROVED** [review]
`firefly_client.py` and `env.py` clean up looks good. Documentation updates look good.

I am glad we are staying up-to-date on packaging.

**Pull request merged**